### ISSUE TITLE
[Assert] Reopening a Script Canvas graph for the first time throws an Assert to the Console

### ISSUE THREAD

**Issue by LB-TomaszSerwanski** [o3de/o3de#9421]
**Describe the bug**
The following Assert is thrown in the Console after reopening a Script Canvas graph after it has been closed:

**File:**
`D:/Git/o3de/Gems/ScriptCanvas/Code/Editor/View/Windows/MainWindow.cpp`

**Condition:**
```
'void __cdecl ScriptCanvasEditor::MainWindow::SetActiveAsset(const class ScriptCanvasEditor::SourceHandle &)'
 A graph was opened, but a tab was not created for it.
```

**Line:** `2471`

Please refer to the attached video for more details.

**Steps to reproduce**
1. Open Script Canvas.
2. Create and save any valid graph.
3. Close the graph.
4. Reopen the same graph.

**Expected behavior**
No Asserts are being thrown to the Console.

**Actual behavior**
An Assert is thrown to the Console.

**Screenshots/Video**

https://user-images.githubusercontent.com/86953108/167097220-5b7bdade-41d9-4400-bb21-5c100b6f210b.mp4

**Found in Branch**
Stabilization_2205 (32e9143)

**Desktop/Device:**
 - Device: PC
 - OS: Windows
 - Version 10
 - CPU AMD Ryzen 5 3600
 - GPU Nvidia RTX 2060 SUPER
 - Memory 16GB

**Attachments**
[Assert.txt](https://github.com/o3de/o3de/files/8638685/Assert.txt)

**Comment by carlitosan** [comment]
Unable to reproduce, likely fixed with recent changes to the SourceHandle and MainWindow.

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on the Development (8493375) build.
